.travis.yml

```diff
@@ -80,3 +72,4 @@ cache:
   directories:
     - $HOME/.pyenv
     - $HOME/.cache/pip
+    - $HOME/build/mitmproxy/.tox
```

Comment:
tox does a very good job of recreating environments if necessary.
I actually want to get rid of running the whole test suite twice just because a handful of tests might be different. Finding those tests and expanding them to cover both OpenSSL versions should be much cleaner and a lot easier to maintain.